sdk/eventhubs/azure_messaging_eventhubs/README.md

```diff
@@ -1,277 +1,233 @@
-```
+The example shown below uses a `DefaultAzureCredential`, which is appropriate for most local development environments. Additionally, we recommend using a managed identity for authentication in production environments. You can find more information on different ways of authenticating and their corresponding credential types in the [Azure Identity] documentation.
 
-That should output something like:
-
-```json
-{
-    "createdAt": "2023-08-10T21:02:07.62Z",
-    "id": "/subscriptions/<your subscription>/resourceGroups/<your group name>/providers/Microsoft.EventHub/namespaces/<your namespace name>/eventhubs/<your eventhub name>",
-    "location": "westus",
-    "messageRetentionInDays": 7,
-    "name": "<your eventhub name>",
-    "partitionCount": 4,
-    "partitionIds": ["0", "1", "2", "3"],
-    "resourceGroup": "<your group name>",
-    "retentionDescription": {
-        "cleanupPolicy": "Delete",
-        "retentionTimeInHours": 168
-    },
-    "status": "Active",
-    "type": "Microsoft.EventHub/namespaces/eventhubs",
-    "updatedAt": "2023-08-10T21:02:16.29Z"
-}
-```
+The `DefaultAzureCredential` will automatically pick up on an Azure CLI authentication. Ensure you are logged in with the Azure CLI:
 
-### Authenticate the client
+```azurecli
+az login
+```
 
-Event Hubs clients are created using a credential from the [Azure Identity package][azure_identity_pkg], like [DefaultAzureCredential][default_azure_credential].
+Instantiate a `DefaultAzureCredential` to pass to the client. The same instance of a token credential can be used with multiple clients if they will be authenticating with the same identity.
 
-# Key concepts
+### Create an Event Hubs message producer on an Event Hubs instance
 
-An Event Hub [**namespace**](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#namespace) can have multiple event hubs.
-Each Event Hub instance, in turn, contains [**partitions**](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#partitions) which store events.
+```rust no_run
+use azure_identity::DefaultAzureCredential;
+use azure_messaging_eventhubs::ProducerClient;
 
-<!-- NOTE: Fix dead links -->
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let host = "<EVENTHUBS_NAMESPACE_HOST>";
+    let eventhub = "<EVENTHUB_NAME>";
 
-Events are published to an Event Hub instance using an [event publisher](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#event-publishers). In this package, the event publisher is the [ProducerClient]()
+    // Create new credential
+    let credential = DefaultAzureCredential::new()?;
 
-Events can be consumed from an Event Hub instance using an [event consumer](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#event-consumers).
+    // Create and open new Producer Client
+    let producer = ProducerClient::new(
```

Comment:
This sample code won't compile because it's out-of-date :(. Sorry about that, but based on feedback from the bug bash, I'm redesigning how producers and consumers are instantiated.